stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -16,27 +22,83 @@ class StockReleaseChannel(models.Model):
         ),
     )
 
+    @delivery_date_generator(step="customer")
+    def _next_delivery_date_partner_delivery_window(self, delivery_date, partner):
+        """Get the next valid delivery date respecting customer delivery window.
+
+        The delivery date must be when the customer is open.
+        """
+        self.ensure_one()
+        partner.ensure_one()
+        if not self.respect_partner_delivery_time_windows:
+            while True:
```

Comment:
I tend to prefer strings over True in while loops.
It's self-documenting, which is good
```suggestion
            while "user hasn't done something":
```
Here's the thing, I have no idea what this while loop is waiting for, and this would be helpful :pray:

Reply:
I added this comment:
```
A delivery date generator needs to provide the earliest valid date
starting from the received date. It can be called multiple times with a
new date to validate.
```
it's ending when it's closed by caller, that's why there is this `while True`